pkg/registry/apps/correlations/legacy/sql.go

```diff
@@ -0,0 +1,107 @@
+package legacy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	correlations "github.com/grafana/grafana/apps/correlations/pkg/apis/correlation/v0alpha1"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"github.com/grafana/grafana/pkg/storage/legacysql"
+	"github.com/grafana/grafana/pkg/storage/unified/sql/sqltemplate"
+)
+
+type LegacySQL struct {
```

Comment:
🤔  we could skip the custom SQL... if we avoid joins and lookup the datasource type manually